### DISCUSSION TITLE
rio.to_raster() and unmanaged memory problems with dask

### DISCUSSION THREAD

**Discussion by martin-git** [corteva/rioxarray#677] [Q&A]
I am trying to do simple processing on a very large Geotiffs (32GB+) using dask.distributed multiprocessing but I am getting massive issues with unmanaged memory filling up RAM. 

What I notice is that, no matter how i set it up (worker numbers, thread numbers, memory limits, chunk sizes,,... ) each worker starts to fill up 60% of the available ram on the machine with unmanaged memory. Setting a memory_limit <32GB only leads to memory warnings and crashes. decreasing the memory target fraction for dask has no effect either.

So far the only working way I have found is to set n_workers=1, so the 32gb of unmanaged memory fit into ram. 
The issue sounds a lot like what was discussed here:
[https://github.com/dask/distributed/issues/6232](url)

Here is a small snipped of the code:

```
from dask.distributed import Client, Lock
import rioxarray as riox
import numpy as np
from pathlib import Path


client = Client(n_workers=1, threads_per_worker=2, memory_limit="55GB", memory_target_fraction=0.3, processes=True)

rxds_proba_2018 = riox.open_rasterio(fp_proba_2018, chunks={"band":11, "x":chunksize_divisible[1], "y":chunksize_divisible[2]}, cache=False)
rxds_proba_2021 = riox.open_rasterio(fp_proba_2021, chunks={"band":11, "x":chunksize_divisible[1], "y":chunksize_divisible[2]}, cache=False)

# computations
rxds_change = (rxds_proba_2021.astype(np.int16) - rxds_proba_2018.astype(np.int16))
rxds_change_out = rxds_change.rio.to_raster(dp_workdir/"change.tif", tiled=True, compute=True, lock=Lock("rio-write", client=client))

````
Typical situation during processing:

![image](https://github.com/corteva/rioxarray/assets/4502797/a638bb1b-d398-4a7b-a544-850ff6c7354d)

Is this a known issue or am i overseeing something?

**Comment by jessjaco** [comment]
I don't know how the data are stored, but have you tried a different chunk size along the band dimension? (Might be triggering a rechunking operation). I usually use a chunk size of 1 along a band dimension.
